gensim/models/sent2vec.py

```diff
@@ -0,0 +1,453 @@
+
+        `dropoutk` = Number of ngrams dropped when training a sent2vec model. Default is 2.
+        """
+
+        random.seed(seed)
+        np.random.seed(seed)
+        self.negpos = 1
+        self.loss = 0.0
+        self.nexamples = 1
+        self.negative_table_size = 10000000
+        self.negatives = []
+        self.vector_size = vector_size
+        self.lr = lr
+        self.lr_update_rate = lr_update_rate
+        self.epochs = epochs
+        self.min_count = min_count
+        self.neg = neg
+        self.word_ngrams = word_ngrams
+        self.loss_type = loss_type
+        self.bucket = bucket
+        self.t = t
+        self.minn = minn
+        self.maxn = maxn
+        self.dropoutk = dropoutk
+
+    def negative_sampling(self, target, lr):
+        loss = 0.0
+        self.grad = np.zeros(self.vector_size)
+        for i in range(self.neg + 1):
+            if i == 0:
+                loss += self.binary_logistic(target, True, lr)
+            else:
+                loss += self.binary_logistic(self.get_negative(target), False, lr)
+        return loss
+
+    def sigmoid(self, val):
+        return 1.0 / (1.0 + np.exp(-val))
+
+    def binary_logistic(self, target, label, lr):
+        score = self.sigmoid(np.dot(self.wo[target], self.hidden))
+        alpha = lr * (float(label) - score)
+        self.grad += self.wo[target] * alpha
+        self.wo[target] += self.hidden * alpha
+        if label is True:
+            return -np.log(score)
+        else:
+            return -np.log(1.0 - score)
+
+    def init_table_negatives(self, counts):
+        z = 0.0
+        for i in range(len(counts)):
+            z += counts[i] ** 0.5
+        for i in range(len(counts)):
+            c = counts[i] ** 0.5
+            for j in range(int(c * self.negative_table_size / z) + 1):
+                self.negatives.append(i)
+        random.shuffle(self.negatives)
+
+    def get_negative(self, target):
+        while True:
+            negative = self.negatives[self.negpos]
+            self.negpos = (self.negpos + 1) % len(self.negatives)
+            if target != negative:
+                break
+        return negative
+
+    def update(self, input_, target, lr):
+        assert(target >= 0)
+        assert(target < self.dict.size)
+        if len(input_) == 0:
+            return
+        self.hidden = np.zeros(self.vector_size)
+        for i in input_:
+            self.hidden += self.wi[i]
+        self.hidden *= (1.0 / len(input_))
+        self.loss += self.negative_sampling(target, lr)
+        self.nexamples += 1
+        self.grad *= (1.0 / len(input_))
+        for i in input_:
+            self.wi[i] += self.grad
+
+    def train(self, sentences):
+        """
+        Update the model's neural weights from a sequence of sentences.
+        For Sent2Vec, each sentence must be a list of unicode strings.
+        """
+
+        logger.info("Creating dictionary...")
```

Comment:
Change this method according to w2v (train in __init__ if `sentences` is provided and so on)

Reply:
Done. Kindly verify in the current commit.